src/python/pants/backend/helm/resolve/remotes.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass
+from typing import Any, Generator, cast
+
+from pants.option.parser import Parser
+from pants.util.frozendict import FrozenDict
+from pants.util.memo import memoized_method
+
+ALL_DEFAULT_HELM_REGISTRIES = "<ALL DEFAULT HELM REGISTRIES>"
+
+OCI_REGISTRY_PROTOCOL = "oci://"
+
+
+class InvalidHelmRegistryAddress(ValueError):
+    def __init__(self, alias: str, address: str) -> None:
+        super().__init__(
+            f"The registry '{alias}' needs to have a valid OCI address URL "
+            f"(using protocol '{OCI_REGISTRY_PROTOCOL}'). The given address was instead: {address}"
+        )
+
+
+class InvalidHelmClassicRepositoryAddress(ValueError):
+    def __init__(self, alias: str, address: str) -> None:
+        super().__init__(
+            f"The classic repository '{alias}' can not have an OCI address URL "
+            f"(using protocol '{OCI_REGISTRY_PROTOCOL}'). The given address was instead: {address}"
+        )
+
+
+class HelmRemoteAliasNotFoundError(ValueError):
+    def __init__(self, alias: str) -> None:
+        super().__init__(f"There is no Helm remote configured with alias: {alias}")
+
+
+@dataclass(frozen=True)
+class HelmRemote(ABC):
```

Comment:
For my own education, what does `ABC` provide as a base class here?

Reply:
simply marking this class as abstract, but it may be my limited knowledge on Python the one at play here